packages/lexical/src/nodes/LexicalTextNode.ts

```diff
@@ -450,6 +450,9 @@ export class TextNode extends LexicalNode {
     const tag = outerTag === null ? innerTag : outerTag;
     const dom = document.createElement(tag);
     let innerDOM = dom;
+    if (outerTag === 'code') {
```

Comment:
Probably safer to compare against the property value 
```suggestion
    if (this.hasFormat('code')) {
```